src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -67,10 +72,19 @@ private static bool IsAotMethod()
 
         internal static string GetArchitecture() => GetCurrentPlatform().ToString();
 
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatformGuard("windows")]
+#endif
         internal static bool IsWindows() => IsOSPlatform(OSPlatform.Windows);
```

Comment:
Can you change these methods to call `OperatingSystem.Is***` on .NET 6+? Same with `IsWasm` above.

Reply:
Would those flow the platform compatibility annotations? I tried to touch as little existing code as possible, these hash defines are just fixing new compiler warnings for the `net6.0` TFM.

Reply:
No they wouldn't, but the check would be optimized by the JIT.

This (and the other review comment) is some low-hanging fruit we can address now that the library targets .NET 6. I understand if you want to leave it for another time and do here the minimum to reduce the dependency tree.